### ISSUE TITLE
Add better error message for failed request

### ISSUE THREAD

**Issue by DarkKronicle** [trumank/mint#117]
A mod I was playing with got taken down on modio, so updating cache caused a 403. Looking at logs there was only this error, with no information on which mod/request caused it. By going through the last ~20 or so requests I was able to manually check, but it would be helpful if the error message said which request it failed on.

```
ERROR mint::gui::message: modio::Error {
    kind: Status(
        403,
    ),
    source: Error {
        code: 403,
        error_ref: 15024,
        message: "The mod ID you have included in your request could not be found.",
        errors: None,
    },
}
```

The mod in question: https://mod.io/g/drg/m/better-bad-terrain#3436106

**Comment by jieyouxu** [comment]
cc @monsoonification

**Comment by jieyouxu** [comment]
Note that you should also:

- Clear the `app.request_failed_mod_id` (or whatever you name it) on App as soon as the user clicks the integrate / install mods button.
- Get the mod id from ur propagated error information, compare it with the available mod information in the GUI, then render an error icon next to the appropriate entry.

**Comment by trumank** [comment]
Implemented in #138
